### DISCUSSION TITLE
extendMarkdown() missing 1 required positional argument: 'md_globals'

### DISCUSSION THREAD

**Discussion by Jachimo** [obsidian-html/obsidian-html#427] [Q&A] [answered]
I'm not sure this is a bug or if it's user error, so I thought I'd bring it up here before creating an issue…

After grabbing the latest code and installing all dependencies (I'm running inside PyCharm), I wanted to do a quick smoketest and created a trivial `config.yml` file, pointing to the included test data inside `ci/test_vault/`:

```YAML
obsidian_folder_path_str: 'ci/test_vault/'
obsidian_entrypoint_path_str: 'ci/test_vault/entrypoint.md'
```

I then tried to run the module using that config (this is inside a venv shell with all dependencies installed using pip):

```Bash
$ cd obsidian-html
$ python -m obsidianhtml -i jachimo-test.yml
[... various output]
> COMPILING HTML FROM MARKDOWN CODE (/Volumes/HD/Users/jachimo/src/obsidian-html/output/md/index.md)
[ traceback ]
TypeError: extendMarkdown() missing 1 required positional argument: 'md_globals'
```
This error seems to happen regardless of whether I'm running the latest HEAD or the 1.0.1 release tag code, and it's not clear to me that I'm doing anything wrong.  Is there something I'm missing? It seems like what I'm doing ought to work.

I also get the same error when trying to use the "run" command (`python -m obsidianhtml run -f ci/test_vault/entrypoint.md`) with the same `config.yml` file sitting in the working dir.

**Comment by dwrolvink** [comment]
This suggests you are using an older version of markdown.
What is the output for `pip show markdown` for you?

If it's below 3.4.1 try and see if you can update it to that version.

**Reply by Jachimo** [reply]
I get:

```Text
$ pip show markdown
Name: Markdown
Version: 3.4.1
Summary: Python implementation of Markdown.
Home-page: https://Python-Markdown.github.io/
Author: Manfred Stienstra, Yuri takhteyev and Waylan limberg
Author-email: python.markdown@gmail.com
License: BSD License
Location: /Volumes/Raptor/Users/jtuttle/src/obsidian-html/venv/lib/python3.9/site-packages
Requires: importlib-metadata
Required-by: obsidianhtml, obsidianhtml-md-mermaid-fork
```
That seems consistent with the latest release on https://github.com/Python-Markdown/markdown/tags

**Comment by dwrolvink** [comment]
Hmm installing an older version of markdown I get a different error when running the latest code...

- What do you get when you run `obsidianhtml version`?
- Have you run `pip install .` while in the cloned folder at least once?
- Does the issue also happen if you run `pip install obsidianhtml; obsidianhtml run -f ci/test_vault/entrypoint.md` ?

Also, can you provide the traceback? I'm mostly curious which lines it hits on, perphaps it's a specific module.

**Reply by Jachimo** [reply]
I've run `pip install .` in the cloned repo, and just re-ran it to be sure. 
See output in [obsidian-markdown_pip-install.txt](https://github.com/obsidian-html/obsidian-html/files/9667852/obsidian-markdown_pip-install.txt), which looks fine to me.

Running either `obsidianhtml version` or `python -m obsidianhtml version` throws the `md_globals` error.

This is the output I'm getting: [error.log](https://github.com/obsidian-html/obsidian-html/files/9667928/error.log)

FWIW, my `python --version` is Python 3.9.13, and `pip --version` is 22.2.2.

**Comment by dwrolvink** [comment, marked as answer]
The error points to ConvertMarkdownPageToHtmlPage on `__init__.py" line 234`, which does not exist in that location anymore in the latest versions. This would suggest that you are on an obsidianhtml version earlier than v3.2.0. (Compare [current](https://github.com/obsidian-html/obsidian-html/blob/master/obsidianhtml/__init__.py) with [v3.1.0](https://github.com/obsidian-html/obsidian-html/blob/v3.1.0/obsidianhtml/__init__.py))

The markdown fix was [introduced in version 3.0.2](https://github.com/obsidian-html/obsidian-html/pull/293/files). See also #291.

What does `pip show obsidianhtml` tell you, and when does it say after `pip install obsidianhtml --upgrade`? 

If you are in the code folder itself, look into `setup.cfg` in the root of the folder, the version is listed on line 3.

Please upgrade to the latest version and use `obsidianhtml` instead of `python -m obsidianhtml` commands from then on, so that I know what kind of code we are troubleshooting.

If you are at the latest version, and still get the same error on `__init__.py" line 234` something very weird is going on. Possibly some shenanigans with virtualenvs/wrong git checkouts?

**Reply by Jachimo** [reply]
Ahh, I think that's the answer!

```Bash
$ pip show obsidianhtml
Name: obsidianhtml
Version: 1.0.1
[...]
```

And version 1.0.1 is also what's shown in my copy of `setup.cfg`, so that makes sense. It looks like I had a very borked checkout.

I just blew away the local copy and re-cloned, and this time it looks like (running `git log`) I have all the commits right up to the latest HEAD (bfe1ae5d74c188e129f15842f625ee046b3e661c). After re-running `pip install .` to grab dependencies, it now says `Successfully installed obsidianhtml-3.2.3 whoosh-2.7.4`, and `python obsidianhtml version` **and** `python -m obsidianhtml version` now report "3.2.3 A".

Running a quick test against ci/test_vault now seems to work properly.

Thanks!

**Reply by dwrolvink** [reply]
Nice! Good luck with using obsidianhtml! Hope it fits your needs.
